hypha/static_src/src/app/src/containers/ReminderContainer/selectors.js

```diff
@@ -0,0 +1,24 @@
+import { createSelector } from 'reselect';
+import initialState from './models';
+
+export const selectFieldsRenderer = state =>
+  state.ReminderContainer ? state.ReminderContainer : initialState;
+
+export const selectReminderContainer = createSelector(selectFieldsRenderer, domain => domain);
+
+export const selectReminders = createSelector(selectReminderContainer, domain => {
+  let reminders = []
+  domain.reminders && domain.reminders.map(reminder => {
+    if(reminders.find(r => r.grouper == reminder.action_type)){
+      const index = reminders.indexOf(reminders.find(r => r.grouper == reminder.action_type))
```

Comment:
We are looping over `reminders` 3 times.

What do you think about using `findIndex` to cut down the loops?

Something like:

```js
const existingReminderIndex = reminders.findIndex(r => r.grouper == reminder.action_type);
// reminder is found.
if (existingReminderIndex != -1) {
  reminders[existingReminderIndex].list.push(reminder)
} else {
  // new reminder.
  reminders.push({ grouper: reminder.action_type, list: [reminder] })
}
```